school/src/main/java/com/hogwarts/school/controller/FacultyController.java

```diff
@@ -18,40 +19,44 @@ public FacultyController(FacultyService facultyService) {
         this.facultyService = facultyService;
     }
 
-    @PostMapping("/addFaculty")
+    @PostMapping
     public ResponseEntity<Faculty> addFaculty(@RequestBody Faculty faculty) {
-        Faculty fcl = facultyService.addFaculty(faculty);
-        return ResponseEntity.ok(fcl);
+        return ResponseEntity.ok(facultyService.createFaculty(faculty));
     }
 
-    @GetMapping("/getFaculty/{facultyId}")
+    @GetMapping("/{facultyId}")
     public ResponseEntity<Faculty> getFaculty(@PathVariable Long facultyId) {
-        Faculty fcl = facultyService.getFaculty(facultyId);
-        if (facultyId == null) {
-            return ResponseEntity.notFound().build();
-        }
-        return ResponseEntity.ok(fcl);
+        return ResponseEntity.ok(facultyService.findFaculty(facultyId));
     }
 
-    @PutMapping()
-    public ResponseEntity<Faculty> updateFaculty(@RequestBody Faculty faculty) {
-        Faculty fcl = facultyService.updateFaculty(faculty);
-        if (fcl == null) {
-            return ResponseEntity.notFound().build();
-        }
-        return ResponseEntity.ok(fcl);
+    @PutMapping("/{facultyId}")
+    public ResponseEntity<Faculty> updateFaculty(@PathVariable Long faultyId, @RequestBody Faculty faculty) {
+        return ResponseEntity.ok(facultyService.updateFaculty(faultyId, faculty));
     }
-    @DeleteMapping("/removeFaculty/{facultyId}")
+
+    @DeleteMapping("/{facultyId}")
     public ResponseEntity<Faculty> removeStudent(@PathVariable Long facultyId) {
-        Faculty fcl = facultyService.removeFaculty(facultyId);
-        if (fcl == null) {
-            return ResponseEntity.notFound().build();
-        }
-        return ResponseEntity.ok(fcl);
+        return ResponseEntity.ok(facultyService.removeFaculty(facultyId));
     }
-    @GetMapping("/getFacultyByColor/{color}")
-    public ResponseEntity<List<Faculty>> getFacultyByColor(@PathVariable String color) {
-        List<Faculty> fcl = facultyService.filteringFacultyByColor(color);
+
+    @GetMapping(params = "color")
+    public ResponseEntity<Collection<Faculty>> getFacultyByColor(@RequestParam String color) {
+        Collection<Faculty> fcl = facultyService.filteringFacultyByColor(color);
         return ResponseEntity.ok(fcl);
     }
+
+    @GetMapping
+    public ResponseEntity<Collection<Faculty>> getAll() {
+        return ResponseEntity.ok(facultyService.findAllFaculty());
+    }
+
+    @GetMapping(params = "colorOrName")
+    public List<Faculty> findByColorOrName(@RequestParam String colorOrName) {
+        return facultyService.findByColorOrName(colorOrName);
+    }
+
+    @GetMapping("/{id}/student")
```

Comment:
```suggestion
    @GetMapping("/{id}/students")
```
не одного же студента отдаём, а все, какие есть